File: app/src/main/java/jgappsandgames/smartreminderslite/priority/PriorityActivity.java
package jgappsandgames.smartreminderslite.priority;

// Java
import java.util.ArrayList;

// Android OS
import android.content.Intent;
import android.os.Bundle;

// Views
import android.view.View;

// Program
import jgappsandgames.smartreminderslite.R;
import jgappsandgames.smartreminderslite.home.FirstRun;

// Save
import jgappsandgames.smartreminderssave.MasterManager;
import jgappsandgames.smartreminderssave.tags.TagManager;
import jgappsandgames.smartreminderssave.tasks.Task;
import jgappsandgames.smartreminderssave.tasks.TaskManager;
import jgappsandgames.smartreminderssave.utility.FileUtility;

/**
 * PriorityActivity
 * Created by joshua on 10/1/17.
 */
public class PriorityActivity extends PriorityActivityInterface {
    // Data
    private ArrayList<Task> ignore_tasks;
    private ArrayList<Task> low_tasks;
    private ArrayList<Task> normal_tasks;
    private ArrayList<Task> high_tasks;
    private ArrayList<Task> stared_tasks;

    // LifeCycle Methods
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // First Run
        if (FileUtility.isFirstRun()) {
            Intent first_run = new Intent(this, FirstRun.class);
            startActivity(first_run);
        } else {
            MasterManager.load(this);
        }

        ignore_tasks = new ArrayList<>();
        low_tasks = new ArrayList<>();
        normal_tasks = new ArrayList<>();
        high_tasks = new ArrayList<>();
        stared_tasks = new ArrayList<>();

        for (String t : TaskManager.tasks) {
            Task task= new Task(t);

            if (task.getType() == Task.TYPE_FLDR) continue;

            if (task.getPriority() == 0) ignore_tasks.add(task);
            else if (task.getPriority() <= 33) low_tasks.add(task);
            else if (task.getPriority() <= 66) normal_tasks.add(task);
            else if (task.getPriority() <= 99) high_tasks.add(task);
            else stared_tasks.add(task);
        }
    }

    @Override
    protected void onResume() {
        super.onResume();

        ignore_adapter = null;
        low_adapter = null;
        normal_adapter = null;
        high_adapter = null;
        stared_adapter = null;

        setAdapter();
        setTitle();
    }

    @Override
    protected void onPause() {
        super.onPause();

        save();
    }

    // Click Listeners
    @Override
    public void onClick(View view) {
        if (view.equals(down)) {
            switch (position) {
                case 1:
                    break;

                case 2:
                    position = 1;
                    setAdapter();
                    setTitle();
                    break;

                case 3:
                    position = 2;
                    setAdapter();
                    setTitle();
                    break;

                case 4:
                    position = 3;
                    setAdapter();
                    setTitle();
                    break;

                case 5:
                    position  = 4;
                    setAdapter();
                    setTitle();
                    break;
            }
        } else if (view.equals(up)) {
            switch (position) {
                case 1:
                    position = 2;
                    setAdapter();
                    setTitle();
                    break;

                case 2:
                    position = 3;
                    setAdapter();
                    setTitle();
                    break;

                case 3:
                    position = 4;
                    setAdapter();
                    setTitle();
                    break;

                case 4:
                    position = 5;
                    setAdapter();
                    setTitle();
                    break;

                case 5:
                    break;
            }
        }
    }

    // Task Changed Listeners
    @Override
    public void onTaskChanged() {
        onResume();
    }

    // Private Class Methods
    private void setTitle() {
        switch (position) {
            case 1:
                setTitle("Ignore");
                down.setText("");
                up.setText(R.string.low);
                return;

            case 2:
                setTitle("Low Priority (Default)");
                down.setText(R.string.ignore);
                up.setText(R.string.normal);
                return;

            case 3:
                setTitle("Normal Priority");
                down.setText(R.string.low);
                up.setText(R.string.high);
                return;

            case 4:
                setTitle("High Priority");
                down.setText(R.string.normal);
                up.setText(R.string.stared);
                return;

            case 5:
                setTitle("Stared Tasks");
                down.setText(R.string.high);
                up.setText("");
                break;
        }
    }

    private void setAdapter() {
        switch (position) {
            case 1:
                if (ignore_adapter == null) ignore_adapter = new PriorityAdapter(this, ignore_tasks);
                tasks.setAdapter(ignore_adapter);
                return;

            case 2:
                if (low_adapter == null) low_adapter = new PriorityAdapter(this, low_tasks);
                tasks.setAdapter(low_adapter);
                return;

            case 3:
                if (normal_adapter == null) normal_adapter = new PriorityAdapter(this, normal_tasks);
                tasks.setAdapter(normal_adapter);
                return;

            case 4:
                if (high_adapter == null) high_adapter = new PriorityAdapter(this, high_tasks);
                tasks.setAdapter(high_adapter);
                return;

            case 5:
                if (stared_adapter == null) stared_adapter = new PriorityAdapter(this, stared_tasks);
                tasks.setAdapter(stared_adapter);
                break;
        }
    }

    @Override
    public void save() {
        MasterManager.save();
    }
}

File: app/src/main/java/jgappsandgames/smartreminderslite/sort/priority/PriorityActivity.kt
package jgappsandgames.smartreminderslite.sort.priority

// Android OS
import android.app.Activity
import android.content.Intent

// Views
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.BaseAdapter

// App
import jgappsandgames.smartreminderslite.R
import jgappsandgames.smartreminderslite.adapter.TaskAdapterInterface
import jgappsandgames.smartreminderslite.holder.TaskFolderHolder
import jgappsandgames.smartreminderslite.home.FirstRun
import jgappsandgames.smartreminderslite.utility.OptionsUtility

// Save
import jgappsandgames.smartreminderssave.MasterManager
import jgappsandgames.smartreminderssave.priority.PriorityManager
import jgappsandgames.smartreminderssave.tasks.Task
import jgappsandgames.smartreminderssave.utility.FileUtility
import kotlinx.android.synthetic.main.activity_priority.*

/**
 * PriorityActivity
 * Created by joshua on 12/14/2017.
 *
 * Priority[PriorityActivityInterface, PriorityActivity]
 * A View for Viewing Tasks Based on Their Priority
 *
 *     PriorityActivityInterface: View and Application Focus
 *     PriorityActivity: Data and User Input
 */
class PriorityActivity:
        Activity(), TaskFolderHolder.OnTaskChangedListener {
    // Data ----------------------------------------------------------------------------------------
    private var position: Int = 3

    // Adapters ------------------------------------------------------------------------------------
    private var ignore: BaseAdapter? = null
    private var low: BaseAdapter? = null
    private var normal: BaseAdapter? = null
    private var high: BaseAdapter? = null
    private var stared: BaseAdapter? = null

    // LifeCycle Methods ---------------------------------------------------------------------------
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_priority)

        // Handle Data
        FileUtility.loadFilePaths(this)
        if (FileUtility.isFirstRun()) startActivity(Intent(this, FirstRun::class.java))
        else MasterManager.load()

        // Click Listeners
        priority_down.setOnClickListener {
            moveDown()
        }

        priority_up.setOnClickListener {
            moveUp()
        }
    }

    override fun onResume() {
        super.onResume()

        // Load Priority Manager
        PriorityManager.create()

        // Reset Adapters
        ignore = null
        low = null
        normal = null
        high = null
        stared = null

        // Set View
        setTitle()
    }

    override fun onPause() {
        super.onPause()
        save()
    }

    // Menu Methods --------------------------------------------------------------------------------
    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_auxilary, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return OptionsUtility.onOptionsItemSelected(this, item, object: OptionsUtility.Save {
            override fun save() {
                this@PriorityActivity.save()
            }
        })
    }

    // Task Listeners ------------------------------------------------------------------------------
    override fun onTaskChanged() {
        onResume()
    }

    // Private Class Methods -----------------------------------------------------------------------
    private fun setTitle() {
        when (position) {
            1 -> {
                title = getString(R.string.ignore)
                priority_down.text = ""
                priority_up.setText(R.string.low)
                if (ignore == null) ignore = PriorityAdapter(this, PriorityManager.getIgnored())
                priority_tasks.adapter = ignore
                return
            }

            2 -> {
                title = getString(R.string.low_priority_default)
                priority_down.setText(R.string.ignore)
                priority_up.setText(R.string.normal)
                if (low == null) low = PriorityAdapter(this, PriorityManager.getLow())
                priority_tasks.adapter = low
                return
            }

            3 -> {
                title = getString(R.string.normal_priority)
                priority_down.setText(R.string.low)
                priority_up.setText(R.string.high)
                if (normal == null) normal = PriorityAdapter(this, PriorityManager.getNormal())
                priority_tasks.adapter = normal
                return
            }

            4 -> {
                title = getString(R.string.high_priority)
                priority_down.setText(R.string.normal)
                priority_up.setText(R.string.stared)
                if (high == null) high = PriorityAdapter(this, PriorityManager.getHigh())
                priority_tasks.adapter = high
                return
            }

            5 -> {
                title = getString(R.string.stared_tasks)
                priority_down.setText(R.string.high)
                priority_up.text = ""
                if (stared == null) stared = PriorityAdapter(this, PriorityManager.getStared())
                priority_tasks.adapter = stared
            }
        }
    }

    private fun moveUp() {
        when (position) {
            1 -> {
                position = 2
                priority_down.visibility = View.VISIBLE
                setTitle()
            }

            2 -> {
                position = 3
                setTitle()
            }

            3 -> {
                position = 4
                setTitle()
            }

            4 -> {
                position = 5
                priority_up.visibility = View.INVISIBLE
                setTitle()
            }
        }
    }

    /**
     * MoveDown
     *
     * Called To Move The Current Selection Down By One
     */
    private fun moveDown() {
        when (position) {
            2 -> {
                position = 1
                priority_down.visibility = View.INVISIBLE
                setTitle()
            }

            3 -> {
                position = 2
                setTitle()
            }

            4 -> {
                position = 3
                setTitle()
            }

            5 -> {
                position = 4
                priority_up.visibility = View.VISIBLE
                setTitle()
            }
        }
    }

    // Parent Overrides ----------------------------------------------------------------------------
    fun save() {
        MasterManager.save()
    }

    // Internal Classes ----------------------------------------------------------------------------
    class PriorityAdapter(activity: PriorityActivity, tasks: ArrayList<Task>):
            TaskAdapterInterface(activity, activity, tasks)
}